core/serialization_v6.c

```diff
@@ -973,6 +1153,27 @@ uint32_t chunk_v6_read_shape(Stream *s,
         cursor = (void *)((uint8_t *)cursor + 1);
         totalSizeRead += 1; // size of chunk id
         switch (chunkID) {
+            case P3S_CHUNK_ID_SHAPE_ID: {
+                memcpy(shapeId, cursor, sizeof(*shapeId));
```

Comment:
Use `sizeof(uint32_t)` for clarity, the way we advance cursor is also based on the type. Overall, please use `sizeof(type)` in every similar line below